wgpu/backends/wgpu_native/_api.py

```diff
@@ -218,7 +230,41 @@ def request_adapter(
             else:
                 logger.warning(f"Forcing backend: {force_backend} ({backend})")
 
-        # ----- Request adapter
+        # Get keywords for adapter selection
+        gpu_keywords = [kw.strip() for kw in (gpu_preference or "").split(",")]
+        gpu_keywords = [kw.lower() for kw in gpu_keywords if kw]
+
+        if gpu_keywords:
+
+            # Get all adapters and select best one based on keywords
+            adapters = self._enumerate_adapters()
+            if force_backend:
+                gpu_keywords.insert(0, force_backend.lower())
+            # Score all adapters
+            for adapter in adapters:
+                adapter_info_str = str(adapter.request_adapter_info()).lower()
+                adapter._score = 0
+                for i, keyword in enumerate(gpu_keywords):
+                    if keyword in adapter_info_str:
+                        adapter._score += len(gpu_keywords) - i
+            # Prioritize. For equal scores, the original order is maintained.
+            adapters.sort(key=lambda adapter: -adapter._score)
```

Comment:
I just realized it's common to have compute nodes with ex. 4 of the same GPU. What do you think about also allowing `gpu_preference` to be an `int` which maps directly to the index of adapters returned from `enumerate_adapters()`, assuming that `enumerate_adapters` always returns GPUs in the same order? I can test this out tomorrow (Wednesday)!

Reply:
Oh dear, I had not considered that yet. I wonder what the `request_adapter_info()` dicts of these look like, and whether it's possible to discern them ... Would be great if you can post the result!

I find the idea of using an int a bit scary, because it will break down so easily. If you add a gpu, if one disconnects for some reason, if you run it on another machine, if wgpu-core lists more or less adapters, or decides to change the order ...